pkg/armhelpers/azureclient.go

```diff
@@ -157,6 +157,26 @@ func NewAzureClientWithClientSecret(env azure.Environment, subscriptionID, clien
 	return getClient(env, subscriptionID, tenantID, armSpt, graphSpt), nil
 }
 
+// NewAzureClientWithClientSecretExternalTenant returns an AzureClient via client_id and client_secret from a tenant
+func NewAzureClientWithClientSecretExternalTenant(env azure.Environment, subscriptionID, tenantID, clientID, clientSecret string) (*AzureClient, error) {
```

Comment:
when is this new function going to get called?

Reply:
It is called from RP, nothing from acs-engine side directly

Reply:
@CecileRobertMichon this function is going to be used by the RP. We add it here as all the Azure client initialization code and functions are in ACS E so it makes sense keep this consolidated in one place with the added advantage of being able to reuse common functions. Can we get a sign off?